src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenSource.cs

```diff
@@ -509,11 +544,11 @@ protected virtual void Dispose(bool disposing)
                 // internal source of cancellation, then Disposes of that linked source, which could
                 // happen at the same time the external entity is requesting cancellation).
 
-                TimerQueueTimer? timer = _timer;
+                ITimer? timer = _timer;
                 if (timer != null)
                 {
                     _timer = null;
-                    timer.Close(); // TimerQueueTimer.Close is thread-safe
+                    timer.Dispose(); // ITimer.Dispose is thread-safe
```

Comment:
We should ensure this assumption about ITimer's thread-safety is documented in remarks on ITimer.